src/plugins/banner/public/components/global_banner.tsx

```diff
@@ -0,0 +1,167 @@
+
+  useEffect(() => {
+    const subscription = actualBannerService.getBannerConfig$().subscribe((config) => {
+      setBannerConfig(config);
+    });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [actualBannerService]);
+
+  // Update the CSS variable with the banner's height
+  useEffect(() => {
+    // Add a smooth transition when changing banner visibility
+    document.documentElement.style.transition = 'padding-top 0.3s ease';
+
+    // If banner is not visible, set height to 0
+    if (!bannerConfig?.isVisible) {
+      document.documentElement.style.setProperty('--global-banner-height', '0px');
+
+      // Reset the transition after a delay
+      setTimeout(() => {
+        document.documentElement.style.transition = '';
+      }, 300);
+
+      return;
+    }
+
+    // Set an initial non-zero value to ensure CSS takes effect
+    document.documentElement.style.setProperty('--global-banner-height', '40px');
+
+    // Reset the transition after a delay
+    setTimeout(() => {
+      document.documentElement.style.transition = '';
+    }, 300);
+
+    // Use ResizeObserver to detect height changes
+    const resizeObserver = new ResizeObserver((entries) => {
+      for (const entry of entries) {
+        // Get height from ResizeObserver
+        const height = entry.contentRect.height;
+
+        // Set the CSS variable with the banner height
+        document.documentElement.style.setProperty('--global-banner-height', `${height}px`);
+
+        // Store the height in localStorage for persistence across page navigation
+        try {
+          localStorage.setItem('opensearch-dashboards-banner-height', `${height}`);
+        } catch (e) {
+          // Ignore localStorage errors
+        }
+      }
+    });
+
+    if (bannerRef.current) {
+      resizeObserver.observe(bannerRef.current);
+
+      // Also set the height directly for immediate effect
+      const height = bannerRef.current.getBoundingClientRect().height;
+      document.documentElement.style.setProperty('--global-banner-height', `${height}px`);
+    }
+
+    return () => {
+      resizeObserver.disconnect();
+
+      // Don't reset the height on page navigation - only when the banner is actually being hidden
+      // We can detect this by checking if the banner config is still visible
+      if (!bannerConfig || !bannerConfig.isVisible) {
+        // Use a transition when removing the banner to prevent sudden layout shifts
+        document.documentElement.style.transition = 'padding-top 0.3s ease';
+
+        // Reset the height when banner is removed
+        document.documentElement.style.setProperty('--global-banner-height', '0px');
+
+        // Reset the transition after a delay
+        setTimeout(() => {
+          document.documentElement.style.transition = '';
+        }, 300);
+      }
+    };
+  }, [bannerConfig]);
+
+  // Handle close button click
+  const handleDismiss = () => {
+    // Update the banner config to hide the banner
```

Comment:
Nit: instead of leaving multiple comments, could we rename this to something concise like `hideBanner()`?